scripts/infra-library.sh

```diff
@@ -116,37 +117,82 @@ function save_metadata() {
-#             $3 - test command to run by the kubetest2 tester
-function create_gke_test_cluster() {
+# Sets the current user as cluster admin for the given cluster.
+# Parameters: $1 - cluster context name
+function acquire_cluster_admin_role() {
+  if [[ -z "$(kubectl get clusterrolebinding cluster-admin-binding 2> /dev/null)" ]]; then
+    if [[ "$1" =~ ^gke_.* ]]; then
+      kubectl create clusterrolebinding cluster-admin-binding \
+        --clusterrole=cluster-admin --user="$(gcloud config get-value core/account)"
+    else
+      kubectl create clusterrolebinding cluster-admin-binding-"${USER}" \
+        --clusterrole=cluster-admin --user="${USER}"
+    fi
+  fi
+}
+
+# Create a test cluster and run the tests if provided.
+# Parameters: $1 - cluster provider name, e.g. gke
+#             $2 - extra kubetest2 flags
+#             $3 - extra cluster creation flags
+#             $4 - test command to run by the kubetest2 tester
+function create_test_cluster() {
   # Fail fast during setup.
   set -o errexit
   set -o pipefail
-  local -n _extra_kubetest2_flags=$1
-  local -n _extra_cluster_creation_flags=$2
-  local -n _tester_command=$3
 
   if function_exists cluster_setup; then
     cluster_setup || fail_test "cluster setup failed"
   fi
 
+  case "$1" in
+    gke) create_gke_test_cluster "$2" "$3" "$4" ;;
+    kind) create_kind_test_cluster "$2" "$3" "$4" ;;
+    *) echo "unsupported provider: $1"; exit 1 ;;
+  esac
+
+  local result="$?"
+  # Ignore any errors below, this is a best-effort cleanup and shouldn't affect the test result.
+  set +o errexit
+  set +o pipefail
+  function_exists cluster_teardown && cluster_teardown
+  echo "Artifacts were written to ${ARTIFACTS}"
+  echo "Test result code is ${result}"
+  exit "${result}"
+}
+
+# Create a KIND test cluster with kubetest2 and run the test command.
+# Parameters: $1 - extra kubetest2 flags
+#             $2 - extra cluster creation flags
+#             $3 - test command to run by the kubetest2 tester
+function create_kind_test_cluster() {
+  # TODO(chizhg): implement with kubetest2
+  return 0
+}
+
+# Create a GKE test cluster with kubetest2 and run the test command.
+# Parameters: $1 - extra kubetest2 flags
+#             $2 - extra cluster creation flags
+#             $3 - test command to run by the kubetest2 tester after the cluster is created (optional)
+function create_gke_test_cluster() {
+  local -n _extra_kubetest2_flags=$1
+  local -n _extra_cluster_creation_flags=$2
+
   echo "Cluster will have a minimum of ${E2E_MIN_CLUSTER_NODES} and a maximum of ${E2E_MAX_CLUSTER_NODES} nodes."
   local _kubetest2_flags=(
     "gke"
     "--create-command=${E2E_GKE_COMMAND_GROUP} container clusters create --quiet --enable-autoscaling
       --min-nodes=${E2E_MIN_CLUSTER_NODES} --max-nodes=${E2E_MAX_CLUSTER_NODES}
       --cluster-version=${E2E_CLUSTER_VERSION}
-      --scopes=${E2E_GKE_SCOPES} --enable-basic-auth --no-issue-client-certificate
```

Comment:
Here I remove `--enable-basic-auth`, as `Basic authentication has been removed in Kubernetes 1.19`